tensorflow/python/framework/ops.py

```diff
@@ -5352,7 +5352,7 @@ def control_dependencies(control_inputs):
    A context manager that specifies control dependencies for all
    operations constructed within the context.
   """
-  if context.executing_eagerly():
+  if executing_eagerly_outside_functions():
```

Comment:
Hm now I'm no longer convinced this is correct. We still need to be able to build control dependencies when inside tf.function; there should be another way to not get graph elements.

Reply:
Makes sense, let's continue the discussion in #40564